flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkWriter.java

```diff
@@ -244,34 +291,121 @@ private void handleFullyFailedRequest(
 
     @Override
     public void close() {
-        AWSGeneralUtil.closeResources(httpClient, kinesisClient);
+        if (kinesisClientProvider != null) {
+            kinesisClientProvider.close();
+        } else {
+            AWSGeneralUtil.closeResources(httpClient, kinesisClient);
```

Comment:
IIUC, by having the provider extend `SdkAutoCloseable`, you can pass it as the last parameter to the util for resource cleanup.